docs/developer/price-calculation.mdx

```diff
@@ -0,0 +1,96 @@
+---
+
+:::note
+The calculations described in this section are aplicable both to checkout and order. 
+:::
+
+## Introduction
+
+There are multiple factors affecting the final price of an order.
+In a nutshell, Saleor first seeks the variant price and shipping cost for a specific channel,
+then applies all discounts, and finally calculates taxes.
+
+```mermaid
+sequenceDiagram
+    Undiscounted price->>+Base price: apply catalogue promotion
+    Undiscounted price->>+Base price: apply line-level voucher
+    Undiscounted price->>+Base price: apply manual line discount
+    Base price->>+ Untaxed price: apply order promotion
+    Base price->>+ Untaxed price: apply order-level voucher
+    Base price->>+ Untaxed price: apply manual line discount
+    Untaxed price->>+ Final price: apply taxes
+```
+
+## Undiscounted price
+
+​​Firstly, Saleor is looking for a variant price and shipping cost for a specific channel.
+There is an option to override the prices by providing a `price` argument in
+[orderLineCreateInput](api-reference/orders/inputs/order-line-create-input.mdx),
+[checkoutLineInput](api-reference/checkout/inputs/checkout-line-input.mdx)
+or [checkoutLineUpdateInput](api-reference/checkout/inputs/checkout-line-update-input.mdx).
+In case of checkout mutations, overriding prices can be done only by app with `HANDLE_CHECKOUTS` permission.
+Whether the price was taken directly from the channel or overridden, it is reflected as an undiscounted price.
+
+## Discount application
+
+### Base price
+
+We first apply [`line-level discounts`](developer/discounts/overview.mdx#level) which decrease unit prices, resulting in the so-called `base price`. The application is done in the following order:
+
+`base_unit_price` = `undiscounted_price` - ((`catalogue_promotion` + `line-level_voucher`) or `manual_line_discount`)
+
+`base_shipping_price` = `undiscounted_shipping_price` - `shipping_voucher`
+
+Some of the line discounts are mutually exclusive:
+
+* manual line discount can't be combined and overrides all other line discounts
+* only one voucher can be applied
+
+### Untaxed price
+
+Having line-level discounts applied, we can sum up all the lines and calculate the base subtotal, which will be the basis for applying [`order-level discounts`](developer/discounts/overview.mdx#level):
+
+`untaxed_subtotal` = `base_subtotal` - (`entire_order_voucher` or `order_promotion` or `manual_discount`)
+
+`untaxed_shipping` = `base_shipping_price` - `manual_discount`
+
+All order-level discounts are mutually exclusive and can't be combined.
+
+* The entire order voucher decreases the subtotal price and doesn't affect the shipping price.
+* The order promotion can either decrease the subtotal or grant a gift, which will be added to the order as an extra order line with zero price.
+* The manual discount decreases both subtotal and shipping prices. If the fixed value is provided, Saleor proportionally computes the subtotal and shipping portion.
```

Comment:
Should we also explain this proportion on `Manual Discounts` page and link it here?